examples/text-generation/run_generation.py

```diff
@@ -293,19 +309,46 @@ def generate():
         HabanaProfile.disable()
         # Compilation
         logger.info("Graph compilation...")
+        dyn_prompt_lens = args.simulate_dyn_prompt
         t0 = time.perf_counter()
         # The first three iterations take longer because of graph compilation
-        for _ in range(args.warmup):
-            generate()
+        if dyn_prompt_lens is None or len(set(dyn_prompt_lens)) == 1:
+            for _ in range(args.warmup):
+                if dyn_prompt_lens is None:
+                    print("Warming up", flush=True)
+                    generate(None, args.reduce_recompile)
+                else:
+                    print("Warming up for shape,", dyn_prompt_lens[0], flush=True)
+                    generate(dyn_prompt_lens[0], args.reduce_recompile)
+        else:
+            if args.bucket_size > 0:
+                mn = min(dyn_prompt_lens)
+                mx = max(dyn_prompt_lens)
+
+                def rounder(x):
+                    return int(math.ceil(x / args.bucket_size) * args.bucket_size)
+
+                min_prompt_len = rounder(mn)
+                max_sentence_len = rounder(mx)
+                for _ in range(args.warmup):
+                    lst = list(range(min_prompt_len, max_sentence_len + 1, args.bucket_size))
+                    for sz in lst:
+                        print("Warming up for shape,", sz - 1, flush=True)
+                        generate(sz - 1, args.reduce_recompile)
```

Comment:
Why `sz - 1` and not `sz`? It should a multiple of `bucket_size` no?

Reply:
this is fine. for bucket_size>1 and most values of sz, sz-1 and sz will compile to the same size (some multiple of bucket_size)

There is some reason I forget exactly why, but its better to have sz-1 than sz.

It doesnt need to be multiple of bucket here, it will be rounded up inside